agent/recorder.go

```diff
@@ -111,11 +111,13 @@ func (r *SpanRecorder) loop() error {
 // Sends the spans in the buffer to Scope
 func (r *SpanRecorder) SendSpans() error {
 	r.Lock()
-	defer r.Unlock()
+	spans := r.spans
+	r.spans = nil
+	r.Unlock()
```

Comment:
Maybe we can have this in a separate function that uses `defer r.Unlock()` inside

Reply:
I thought about it but because there is no possibility of panic on those two lines I refused the idea.